src/oscar/apps/basket/views.py

```diff
@@ -12,19 +12,20 @@
 from django.views.generic import FormView, View
 from extra_views import ModelFormSetView
 
-from oscar.apps.basket import signals
 from oscar.core import ajax
 from oscar.core.compat import user_is_authenticated
 from oscar.core.loading import get_class, get_classes, get_model
 from oscar.core.utils import redirect_to_referrer, safe_referrer
 
-Applicator = get_class('offer.utils', 'Applicator')
+Applicator = get_class('offer.applicator', 'Applicator')
 (BasketLineForm, AddToBasketForm, BasketVoucherForm, SavedLineForm) = get_classes(
     'basket.forms', ('BasketLineForm', 'AddToBasketForm',
                      'BasketVoucherForm', 'SavedLineForm'))
 BasketLineFormSet, SavedLineFormSet = get_classes(
     'basket.formsets', ('BasketLineFormSet', 'SavedLineFormSet'))
-Repository = get_class('shipping.repository', ('Repository'))
+Repository = get_class('shipping.repository', 'Repository')
+basket_addition, voucher_addition, voucher_removal = get_classes(
+    'basket.signals', ('basket_addition', 'voucher_addition', 'voucher_removal'))
```

Comment:
For consistency with other signals imports, for instance https://github.com/django-oscar/django-oscar/blob/6ce56ccbc033341a72838b0cb06935d6ba498c75/src/oscar/apps/customer/receivers.py#L7.